folly/test/TraitsTest.cpp

```diff
@@ -98,6 +98,14 @@ TEST(Traits, unset) {
   EXPECT_TRUE(IsRelocatable<F4>::value);
 }
 
+TEST(Traits, triviallyMoveConstructible) {
+  struct TriviallyMoveConstructible {
+    TriviallyMoveConstructible(const TriviallyMoveConstructible&) = delete;
+    TriviallyMoveConstructible(TriviallyMoveConstructible&&) = default;
```

Comment:
This function being defined but never called at runtime is a violation of `-Wunneeded-member-function`, which is being shown internally.

Reply:
That's unfortunate. Is this just for Clang? Do you suggest just pragma push/popping it?

We should also add this flag to `FollyCompilerUNIX.cmake`. Then, there is a chance that Travis would fail on this error too unless it is Clang only (as we only use GCC for Travis).

Reply:
This is a [clang warning](https://releases.llvm.org/7.0.0/tools/clang/docs/DiagnosticsReference.html#wunneeded-member-function) but the build does not fail because of it. This warning violation is (currently) being surfaced as if it were a lint rule violation.